easybuild/easyblocks/c/cuda.py

```diff
@@ -73,11 +59,18 @@ def fetch_sources(self, sources=None, checksums=None):
         else:
             raise EasyBuildError("Architecture %s is not supported for CUDA on EasyBuild", myarch)
 
-        modified_sources = []
-        for source in sources:
-            modified_sources.append(source % {'cudaarch': cudaarch})
+        super(EB_CUDA, self).__init__(*args, **kwargs)
+
+        self.cfg.template_values['cudaarch'] = cudaarch
+        self.cfg.generate_template_values()
 
-        return Binary.fetch_sources(self, modified_sources, checksums)
+    @staticmethod
+    def extra_options():
```

Comment:
@edmondac Please move this back up, above the `__init__` method (that's just a style issue, not a technical requirement)

Reply:
Done